pkg/debugapi/topology.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package debugapi
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/ethersphere/bee/pkg/jsonhttp"
+)
+
+type topologyResponse struct {
+	Topology string `json:"topology"`
+}
+
+func (s *server) topologyHandler(w http.ResponseWriter, r *http.Request) {
+	ms, ok := s.TopologyDriver.(json.Marshaler)
+	if !ok {
+		s.Logger.Error("topology driver cast to json marshaler error")
+		jsonhttp.InternalServerError(w, "topology json marshal interface error")
+		return
+	}
+
+	bytes, err := ms.MarshalJSON()
+	if err != nil {
+		s.Logger.Errorf("topology marshal to json: %v", err)
+		jsonhttp.InternalServerError(w, err)
+		return
+	}
+	jsonhttp.OK(w, topologyResponse{Topology: string(bytes)})
```

Comment:
This is a bit strange. First marshaling into json to serve it as a string in a json encoded body. I think that it would be better just to have jsonhttp.OK(w, ms).

Reply:
:+1: i fixed this in the subsequent PR. if its OK we can just have it fixed in that PR. otherwise i can pull the changeset into this PR